src/Http/AccessToken.php

```diff
@@ -71,4 +130,133 @@ public function user()
     {
         return $this->belongsTo(User::class);
     }
+
+    /**
+     * Filters which tokens are valid at the given date for this particular token type.
+     * Uses the static::$lifetime value by default, can be overridden by children classes.
+     * @param Builder $query
+     * @param Carbon $date
+     */
+    public static function scopeValid(Builder $query, Carbon $date)
+    {
+        if (static::$lifetime > 0) {
+            $query->where('last_activity_at', '>', $date->clone()->subSeconds(static::$lifetime));
+        }
+    }
+
+    /**
+     * Filters which tokens are expired at the given date and ready for garbage collection.
+     * Uses the static::$lifetime value by default, can be overridden by children classes.
+     * @param Builder $query
+     * @param Carbon $date
+     */
+    public static function scopeExpired(Builder $query, Carbon $date)
+    {
+        if (static::$lifetime > 0) {
+            $query->where('last_activity_at', '<', $date->clone()->subSeconds(static::$lifetime));
+        } else {
+            $query->whereRaw('FALSE');
+        }
+    }
+
+    /**
+     * Shortcut to find a valid token.
+     * @param string $token Token as sent by the user. We allow non-string values like null so we can directly feed any value from a request.
+     * @return AccessToken|null
+     */
+    public static function findValid($token): ?AccessToken
+    {
+        return static::query()->whereValid()->where('token', $token)->first();
+    }
+
+    /**
+     * This query scope is intended to be used on the base AccessToken object to query for valid tokens of any type.
+     * @param Builder $query
+     * @param Carbon|null $date
+     */
+    public function scopeWhereValid(Builder $query, Carbon $date = null)
```

Comment:
Something other might want to comment on: my scope parameters.

I made the date a parameter of the scope. I'm not seeing any use case outside of the integration tests, but I figured out it's nice to have, and allows testing the feature without being impacted by query run time at all.

You could theoretically use this to find if a token will still be valid in a given number of minutes.

There's also two set of scope-related methods. `scopeWhereValid()` is the actual Eloquent scope, which is only meant to be called on the base class. `static scopeValid()` is the method that can be customized for each type.

Technically, I think I could actually make the static scope method protected and it should still work. This might be better.

Reply:
If it's not needed externally, `protected` seems better to me.